marklogic-spark-connector/src/main/java/com/marklogic/spark/writer/DocBuilder.java

```diff
@@ -156,8 +160,12 @@ private DocumentWriteOperation buildMainDocument(DocumentInputs inputs) {
 
         AbstractWriteHandle content = inputs.getContent();
         byte[] classificationResponse = inputs.getClassificationResponse();
-        if (classificationResponse != null && inputs.getExtractedText() == null && Format.XML.equals(((BytesHandle) content).getFormat())) {
-            content = appendDocumentClassificationToXmlContent(((BytesHandle) content).get(), inputs.getInitialUri(), classificationResponse);
+        if (classificationResponse != null && inputs.getExtractedText() == null) {
+            if (Format.XML.equals(((BytesHandle) content).getFormat())) {
+                content = appendDocumentClassificationToXmlContent(((BytesHandle) content).get(), inputs.getInitialUri(), classificationResponse);
```

Comment:
Use DOMHelper for this. DOMHelper has an `extractDocument(DWO)` method in it - just override that with an `extractDocument(AbstractWriteHandle)` method and call it. That avoids the cast, and it also simplifies DocBuilder which won't need to depend directly on a DocumentBuilderFactory anymore.